Janitor Submission

Canna71 opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:
https://github.com/Canna71/obsidian-janitor

## Release Checklist
- [ ] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

liamcain commented:
- [updateNoteWithDate](https://github.com/Canna71/obsidian-janitor/blob/328df6683175cab8d9251a6ca1687d67119f377b/main.ts#L141) if possible, I would suggest changing this function to use the CodeMirror Editor instead of `modify`. From what i can tell, you are always modifying an open file, so it should be pssible and it has several advantages: doesn't lose scroll position, properly applies to undo history.

Otherwise everything else looks good, let me know when this is ready to go in.

Canna71 commented:
Thank you @joethei  for your feedbacks, they are helping a lot. You are perfectly right, I should have thought about it!
I modified the code in order to use the Editor when possible. I would leave the possibility to set the expiration date also when not in edit mode (it doesn't even seem to lose scroll position) but of course now I am using the editor when in edit node and I only change the frontmatter part.
I already updated the release with updated code.

Canna71 commented:
Ah, I also tested it on iOS.
